test/e2e/deployment.go

```diff
@@ -175,9 +201,28 @@ func testRollingUpdateDeployment(f *Framework) {
 
 	err = waitForDeploymentStatus(c, ns, deploymentName, replicas, replicas-1, replicas+1, 0)
 	Expect(err).NotTo(HaveOccurred())
+
+	// Verify that the pods were scaled up and down as expected. We use events to verify that.
+	deployment, err := c.Deployments(ns).Get(deploymentName)
+	Expect(err).NotTo(HaveOccurred())
+	waitForEvents(c, ns, deployment, 4)
+	events, err := c.Events(ns).Search(deployment)
+	if err != nil {
+		Logf("error in listing events: %s", err)
+		Expect(err).NotTo(HaveOccurred())
+	}
+	// There should be 4 events, 2 to scale up the new RC and 2 to scale down the old RC.
+	Expect(len(events.Items)).Should(Equal(4))
+	newRC, err := deploymentutil.GetNewRC(*deployment, c)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(newRC).NotTo(Equal(nil))
+	Expect(events.Items[0].Message).Should(Equal(fmt.Sprintf("Scaled up rc %s to 1", newRC.Name)))
+	Expect(events.Items[1].Message).Should(Equal(fmt.Sprintf("Scaled down rc %s to 1", rcName)))
```

Comment:
While this should be generally true, there is no guarantee that it will always be like this.
If the new pod hasnt become ready yet, then old RC will be scaled down to 2, instead of being scaled down to 1.

This is the reason why we have a separate test case for events. With an RC with replicas = 1, it is guaranteed that there will be 2 events - first to scale up the new RC and then to scale down the old RC.